File: 2/a.ts
// const input = await Bun.file("test.txt").text();
const input = await Bun.file("input.txt").text();

const totals = {
  red: 12,
  green: 13,
  blue: 14,
} as Record<string, number>;

let res = 0;

input.split("\n").forEach((line, i) => {
  let isOkBig = true;
  const [first, second] = line.split(": ");
  second.split("; ").forEach((item) => {
    const rev = item.split(", ");
    console.log(rev);
    rev.forEach((item) => {
      const [num, color] = item.split(" ");
      console.log(num, color);
      if (totals[color] < parseInt(num)) {
        // console.log(i + 1, color, parseInt(num));
        isOkBig = false;
      }
    });
  });
  // console.log(isOkBig, i + 1);
  if (isOkBig) {
    const num = parseInt(first.split(" ")[1]);
    console.log(i + 1);
    res += num;
  }
});

console.log(res);

File: 2/a.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	cubeCount := map[string]int{
		"red":   12,
		"green": 13,
		"blue":  14,
	}

	file, _ := os.ReadFile("input.txt")

	res := 0

Main:
	for i, line := range strings.Split(string(file), "\n") {
		_, gamesText, _ := strings.Cut(line, ": ")
		for _, game := range strings.Split(gamesText, "; ") {
			for _, inst := range strings.Split(game, ", ") {
				numberStr, color, _ := strings.Cut(inst, " ")
				number, _ := strconv.Atoi(numberStr)
				if cubeCount[color] < number {
					continue Main
				}
			}
		}
		res += 1 + i
	}
	fmt.Println(res)
}
